cmd/crowdsec-cli/capi.go

```diff
@@ -0,0 +1,147 @@
+			log.Printf("your password : %s", password)
+			_, err = Client.Auth.RegisterWatcher(context.Background(), models.WatcherRegistrationRequest{
+				MachineID: &id,
+				Password:  &password,
+			})
+			if err != nil {
+				log.Errorf("unable to register to API (%s) : %s", Client.BaseURL, err)
+			}
+			log.Printf("Successfully registered to Central API (CAPI)")
+
+			var dumpFile string
+
+			log.Printf("config : %s", spew.Sdump(csConfig.API.Server))
+			if outputFile != "" {
+				dumpFile = outputFile
+			} else if csConfig.API.Server.OnlineClient.CredentialsFilePath != "" {
+				dumpFile = csConfig.API.Server.OnlineClient.CredentialsFilePath
+			} else {
+				dumpFile = ""
+			}
+			apiCfg := csconfig.ApiCredentialsCfg{
+				Login:    id,
+				Password: password.String(),
+				URL:      APIBaseURL,
+			}
+			apiConfigDump, err := yaml.Marshal(apiCfg)
+			if err != nil {
+				log.Fatalf("unable to marshal api credentials: %s", err)
+			}
+			if dumpFile != "" {
+				err = ioutil.WriteFile(dumpFile, apiConfigDump, 0644)
+				if err != nil {
+					log.Fatalf("write api credentials in '%s' failed: %s", dumpFile, err)
+				}
+				log.Printf("API credentials dumped to '%s'", dumpFile)
+			} else {
+				fmt.Printf("%s\n", string(apiConfigDump))
+			}
+
+			log.Warningf("Run 'systemctl reload crowdsec' for the new configuration to be effective")
+		},
+	}
+	cmdCapiRegister.Flags().StringVarP(&outputFile, "file", "f", "", "output file destination")
+	cmdCapi.AddCommand(cmdCapiRegister)
+
+	var cmdCapiStatus = &cobra.Command{
+		Use:   "status",
+		Short: "Check status with the Central API (CAPI)",
+		Args:  cobra.MinimumNArgs(0),
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+			if csConfig.API.Server == nil {
+				log.Fatalln("There is no configuration on 'api_client:'")
+			}
+			if csConfig.API.Server.OnlineClient == nil {
+				log.Fatalf("Please provide credentials for the API in '%s'", csConfig.API.Server.OnlineClient.CredentialsFilePath)
+			}
+			apiclient.BaseURL, err = url.Parse(csConfig.API.Server.OnlineClient.Credentials.URL)
+			if err != nil {
+				log.Fatalf("failed to parse Local API URL %s : %v ", csConfig.API.Server.OnlineClient.Credentials.URL, err.Error())
+			}
+			apiclient.UserAgent = fmt.Sprintf("crowdsec/%s", cwversion.VersionStr())
+			if err := cwhub.GetHubIdx(csConfig.Cscli); err != nil {
+				log.Fatalf("failed to get Hub index : %v", err)
+			}
+			scenarios, err := cwhub.GetUpstreamInstalledScenariosAsString()
+			if err != nil {
+				log.Fatalf("failed to get scenarios : %s", err.Error())
+			}
+			password := strfmt.Password(csConfig.API.Server.OnlineClient.Credentials.Password)
+			t := models.WatcherAuthRequest{
+				MachineID: &csConfig.API.Server.OnlineClient.Credentials.Login,
+				Password:  &password,
+				Scenarios: scenarios,
+			}
+			Client = apiclient.NewClient(nil)
+			resp, err := Client.Auth.AuthenticateWatcher(context.Background(), t)
```

Comment:
do we want to check that `resp.StatusCode == 200` ?

Reply:
done in `Do()`